src/program.cc

```diff
@@ -109,31 +111,31 @@ void quantum_program::add_program(const quantum_program &p) {
 
 void quantum_program::add_if(const quantum_kernel &k, const operation &cond) {
     // phi node
-    quantum_kernel kphi1(k.name+"_if", platform, qubit_count, creg_count);
+    quantum_kernel kphi1(k.name+"_if", platform, qubit_count, creg_count, breg_count);
```

Comment:
adding breg_count makes sense, but I actually never understood why a kernel has its own qubit_count and creg_count, separate from a program

Reply:
Likewise, but suppose you have two independent kernels and want to synthesize them in parallel.
That was the only example I could come up with.
Not convincing though.
It is also very confusing already to have those separate from program and also separate from platform.